docs/reference/reference-deploy/config/master-config-reference.rst

```diff
@@ -4,957 +4,1565 @@
 
-      -  ``hdfs_path``: The prefix path where all checkpoints will be written to and read from. The
-         resources of each checkpoint will be saved in a subdirectory of ``hdfs_path``, where the
-         subdirectory name is the checkpoint's UUID.
+``instance_slots``
+^^^^^^^^^^^^^^^^^^
 
-      -  ``user``: An optional string value that indicates the user to use for all read and write
-         requests. If left unspecified, the default user of the trial runner container will be used.
+The optional number of GPUs for the AWS instance type. This is used in conjunction with the
+``instance_type`` in order to specify types that are not listed in the ``instance_type`` list above.
+Note that some GPUs may not be supported. **WARNING**: *be sure to specify the correct number of
+GPUs to ensure that provisioner launches the correct number of instances.*
 
-   -  ``type: s3``: Checkpoints are stored in Amazon S3.
+``cpu_slots_allowed``
+^^^^^^^^^^^^^^^^^^^^^
 
-      -  ``bucket``: The S3 bucket name to use.
-      -  ``access_key``: The AWS access key to use.
-      -  ``secret_key``: The AWS secret key to use.
-      -  ``prefix``: The optional path prefix to use. Must not contain ``..``. Note: Prefix is
-         normalized, e.g., ``/pre/.//fix`` -> ``/pre/fix``
-      -  ``endpoint_url``: The optional endpoint to use for S3 clones, e.g.,
-         ``http://127.0.0.1:8080/``.
+Whether to allow slots on the CPU instance types. When ``true``, and if the instance type doesn't
+have any GPUs, each instance will provide a single CPU-based compute slot; if it has any GPUs,
+they'll be used for compute slots instead. Defaults to ``false``.
 
-   -  ``type: azure``: Checkpoints are stored in Microsoft's Azure Blob Storage. Authentication is
-      performed by providing either a connection string, or an account URL and an optional
-      credential.
+``spot``
+^^^^^^^^
 
-      -  ``container``: The Azure Blob Storage container name to use.
-      -  ``connection_string``: The connection string for the service account to use.
-      -  ``account_url``: The account URL for the service account to use.
-      -  ``credential``: The optional credential to use in conjunction with the account URL.
+Whether to use spot instances. Defaults to ``false``. See :ref:`aws-spot` for more details.
 
-      Please only specify either ``connection_string`` or the ``account_url`` and ``credential``
-      pair.
+``spot_max_price``
+^^^^^^^^^^^^^^^^^^
 
-   -  ``type: shared_fs``: Checkpoints are written to a directory on the agent's file system. The
-      assumption is that the system administrator has arranged for the same directory to be mounted
-      at every agent host, and for the content of this directory to be the same on all agent hosts
-      (e.g., by using a distributed or network file system such as GlusterFS or NFS).
+Optional field indicating the maximum price per hour that you are willing to pay for a spot
```

Comment:
Nit: seems like `Required.` is used a lot, but `Optional.` isn't used as much. Not sure if there is guidance on styling for this, but I'll leave that up to you.

Reply:
thank you!